aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,994 @@
+        let resource_signer = create_signer_with_capability(&nft_mint_config.signer_cap);
+
+        // Create the destination collection that holds the unique art NFT.
+        create_collection(&resource_signer, collection_name, collection_description, collection_uri, collection_maximum, collection_mutate_config);
+
+        move_to(&resource_signer, CollectionConfig {
+            collection_name,
+            collection_description,
+            collection_maximum,
+            collection_uri,
+            collection_mutate_config,
+            token_name_base,
+            royalty_payee_address,
+            token_description,
+            token_maximum,
+            token_mutate_config: create_token_mutability_config(&token_mutate_config),
+            royalty_points_den,
+            royalty_points_num,
+            tokens: big_vector::new<TokenAsset>(128),
+            public_mint_limit,
+        });
+
+        // Create the source certificate collection and token.
+        let source_collection_name = collection_name;
+        string::append_utf8(&mut source_collection_name, b" - source collection");
+        create_collection(&resource_signer, source_collection_name, collection_description, utf8(b""), collection_maximum, collection_mutate_config);
+
+        let source_token_data_id = create_tokendata(
+            &resource_signer,
+            source_collection_name,
+            // token name
+            utf8(b""),
+            // token description
+            utf8(b""),
+            // token maximum
+            0,
+            // token uri
+            utf8(b""),
+            signer::address_of(&resource_signer),
+            1,
+            0,
+            create_token_mutability_config(
+                &vector<bool>[ false, false, false, false, true ]
+            ),
+            vector<String>[string::utf8(BURNABLE_BY_OWNER)],
+            vector<vector<u8>>[bcs::to_bytes<bool>(&true)],
+            vector<String>[string::utf8(b"bool")],
+        );
+
+        move_to(&resource_signer, SourceToken { token_data_id: source_token_data_id });
+    }
+
+    /// Set the minting and reveal config of this collection.
+    public entry fun set_minting_and_reveal_config(
+        admin: &signer,
+        whitelist_minting_start_time: u64,
+        whitelist_minting_end_time: u64,
+        whitelist_mint_price: u64,
+        public_minting_start_time: u64,
+        public_minting_end_time: u64,
+        public_mint_price: u64,
+        reveal_time: u64,
+    ) acquires NFTMintConfig, WhitelistMintConfig, PublicMintConfig, RevealConfig {
+        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
+        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
+
+        let now = timestamp::now_seconds();
+        // assert that we are setting the whitelist time to sometime in the future
```

Comment:
nit: Why do we want to enforce this? It could be a possible use case for launching with whitelisting already open.

Reply:
we don't want them to be able to update the whitelist and public minting time and price if the minting has already started

Reply:
they need to commit to the time and more importantly price before the minting starts